src/julia.h

```diff
@@ -727,7 +727,13 @@ static const uint8_t PARTITION_FLAG_DEPRECATED     = 0x20;
 // where calling the function itself will provide a (better) deprecation warning/error.
 static const uint8_t PARTITION_FLAG_DEPWARN        = 0x40;
 
+#if defined(_COMPILER_CLANG_) || defined(_COMPILER_GCC_)
 typedef struct __attribute__((aligned(8))) _jl_binding_partition_t {
+#elif defined(_COMPILER_MICROSOFT_)
+typedef struct __declspec(align(8)) _jl_binding_partition_t {
+#else
+typedef struct _jl_binding_partition_t {
+#endif
```

Comment:
Can you provide a separate `#define` that handles the platform-specific logic elsewhere?

That way this can become:
```julia
typedef struct JL_ALIGNED_ATTR(8) _jl_binding_partition_t {
```